### PR TITLE
Move to new packing helper

### PR DESCRIPTION
**Pull request by richfitz**
I'd like to move this into mcstate at some point (so that it can interface nicely with the packer support there) but for now I'm working in dust as we can iterate more easily

### CODE REVIEW

File: R/interface.R [hunk 1]
```diff
@@ -617,3 +621,15 @@ is_uncalled_generator <- function(sys) {
...
+
+## This will move into mcstate eventually, but this will let us pack
+## and unpack state, once we work out what we need to do with it.
+## There are a few options here such as:
+##
+## * give me the indices of logical compartments 'a', 'b' and 'c'
+## * convert this vector/array into a list of vectors/arrays
+packer <- function(packing_state) {
+  i <- lengths(packing_state) == 0
+  mcstate2::mcstate_packer(names(packing_state)[i], packing_state[!i])
```
Comment (richfitz) [inline R/interface.R:634]:
```suggestion
  mcstate2::mcstate_packer(array = packing_state)
```

once https://github.com/mrc-ide/mcstate2/pull/62 is merged, though more likely we'll do


```r
  res$packer_state <- mcstate2::mcstate_packer(array = res$packing_state)
  if (!is.null(res$packing_gradient)) {
    res$packer_gradient <- mcstate2::packer(array = res$packing_gradient)
  }
```

at the top of the changes in interface.R and then drop this wrapper

File: inst/include/dust2/packing.hpp [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+class packing {
+  using mapping_type = std::pair<std::string, std::vector<size_t>>;
+public:
+  packing(std::initializer_list<mapping_type> data)
+    : data_(data) {
+    len_.reserve(data_.size());
+    bool require_nonscalar = false;
+    for (auto& el : data_) {
+      // This check is due to matching the current mcstate interface
```
Comment (richfitz) [inline inst/include/dust2/packing.hpp:21]: This can be relaxed once that other mcstate2 one gets merged

File: tests/testthat/examples/openmp.cpp [hunk 3]
```diff
@@ -30,6 +30,14 @@ class openmp {
     return 1;
```
Comment (richfitz) [inline tests/testthat/examples/openmp.cpp:30]: this method can be removed, though it's harmless

**Review by weshinsley — APPROVED** [review]
Talked through - looks good!

**Pull request merged**